### PR TITLE
dxgi: Add support for ReportLiveObjects

### PR DESCRIPTION
**Pull request by gmodarelli**
Hello, I'm trying to add support for `ReportLiveObjects()`, but I'm getting a segmentation fault upon invocation and I can't figure out why.

I hope opening a PR might move things along :)

image

### CODE REVIEW

**Comment by michal-z** [comment]
I'm not sure but I think that this is old D3D11 API, for D3D12 we should be using https://learn.microsoft.com/en-us/windows/win32/api/d3d12sdklayers/nf-d3d12sdklayers-id3d12debugdevice-reportlivedeviceobjects

**Comment by gmodarelli** [comment]
Perfect, thanks for the info. I'll update the PR :)

**Comment by gmodarelli** [comment]
I think I've managed to get it to work. I've simulated a leak by not deleting a fence and now I get this message in the debug output window 

image

File: libs/zd3d12/src/zd3d12.zig [hunk 1]
```diff
@@ -608,6 +621,11 @@ pub const GraphicsContext = struct {
         for (gctx.cmdallocs) |cmdalloc|
             _ = cmdalloc.Release();
         _ = gctx.wic_factory.Release();
+
+        if (enable_debug_layer) {
```
Comment (michal-z) [inline libs/zd3d12/src/zd3d12.zig:625]:
```suggestion
        if (gctx.debug_device) |debug_device| {
```

File: libs/zd3d12/src/zd3d12.zig [hunk 2]
```diff
@@ -51,6 +51,7 @@ pub const GraphicsContext = struct {
     const max_num_buffered_resource_barriers = 16;
 
     device: *d3d12.IDevice9,
+    debug_device: *d3d12.IDebugDevice,
```
Comment (michal-z) [inline libs/zd3d12/src/zd3d12.zig:54]: Let's make it an optional (`?*d3d12.IDebugDevice`) and set it to `null` when debug layer is disabled. In that way users of `zd3d12` module will know if they can use this field.

**Review by michal-z — CHANGES_REQUESTED** [review]
Thnaks! I've added few comments.

**Comment by gmodarelli** [comment]
Thanks for the review!

File: libs/zd3d12/src/zd3d12.zig [hunk 3]
```diff
@@ -189,6 +190,16 @@ pub const GraphicsContext = struct {
             break :blk device;
         };
 
+        var debug_device: *d3d12.IDebugDevice = undefined;
```
Comment (michal-z) [inline libs/zd3d12/src/zd3d12.zig:193]:
```suggestion
        var debug_device: ?*d3d12.IDebugDevice = null;
```

File: libs/zd3d12/src/zd3d12.zig [hunk 4]
```diff
@@ -539,6 +550,7 @@ pub const GraphicsContext = struct {
 
         return GraphicsContext{
             .device = device,
+            .debug_device = if (enable_debug_layer) debug_device else null,
```
Comment (michal-z) [inline libs/zd3d12/src/zd3d12.zig:553]:
```suggestion
            .debug_device = debug_device,
```

**Review by michal-z — CHANGES_REQUESTED** [review]
Just a super minor change. I think this is simpler because we have only one conditional now.

**Pull request merged**